giraph-core/src/main/java/org/apache/giraph/comm/netty/ChannelRotater.java

```diff
@@ -73,9 +76,9 @@ public synchronized void addChannel(Channel channel) {
    */
   public synchronized Channel nextChannel() {
     if (channelList.isEmpty()) {
-      throw new IllegalArgumentException(
-          "nextChannel: No channels exist for hostname " +
-              address.getHostName());
+      LOG.warn("nextChannel: No channels exist for hostname " +
+        address.getHostName());
+      return null;
```

Comment:
Will the job never fail in this case, then? What happens if the channel keeps getting dropped for some reason?

Reply:
This is called from NettyClient.getNextChannel() which will do a maximum number of retries. However, this counter is reset every time there is a failure, so it is possible that if the channel gets dropped, that we keep retrying. What I will do is make this counter global. The current max is 1000 retries, so that should be enough.